repository/src/main/java/org/alfresco/repo/search/impl/DbOrIndexSwitchingQueryLanguage.java

```diff
@@ -173,7 +194,7 @@ public ResultSet executeQuery(SearchParameters searchParameters)
                 throw new QueryModelException("No query language available");
             }
         case HYBRID:
-            if (!hybridEnabled)
+            if (((solrHybridEnabled != null) && (!solrHybridEnabled)) || (hybridEnabled == null) || (!hybridEnabled))

```

Comment:
I don't know if it's better, but this looks like it can be rewritten as:
```
Boolean.FALSE.equals(solrHybridEnabled) || !Boolean.TRUE.equals(hybridEnabled)
```

Reply:
I'm not sure it's the same. 

eg. what if in a new system, solrHybridEnabled is not set (ie. null) and hybridEnabled is set to true ...